Source/WebCore/css/CSSPrimitiveValue.cpp

```diff
@@ -1175,16 +848,26 @@ static NEVER_INLINE String formatNonfiniteValue(double number, ASCIILiteral suff
 
 NEVER_INLINE String CSSPrimitiveValue::formatNumberValue(ASCIILiteral suffix) const
 {
-    if (!std::isfinite(m_value.number))
-        return formatNonfiniteValue(m_value.number, suffix);
-    return makeString(FormattedCSSNumber::create(m_value.number), suffix);
+    if (hasScalarInPointer()) {
+        auto integer = scalarInteger();
+        if (integer < 1000000)
+            return makeString(static_cast<uint32_t>(integer), suffix);
```

Comment:
What is this test for? It could use a comment/named constant.

Reply:
This constant is due to the six digits of precision that CSS formats with. Numbers larger than this need use the general purpose formatting function. But smaller numbers can just be formatted as integers. I’ll have to revise this if I change the encoding scheme.

Reply:
Added a comment.